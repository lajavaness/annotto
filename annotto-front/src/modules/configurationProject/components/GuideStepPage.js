import 'github-markdown-css'
import { Col, Input, Row, Switch } from 'antd'
import { useDispatch, useSelector } from 'react-redux'
import { useTranslation } from 'react-i18next'
import React, { useCallback, useEffect, useState } from 'react'
import ReactMarkdown from 'react-markdown'

// TODO  this is an example of temporary markdown, we need an example specific to Annotto
import TutorialMarkDown from 'assets/tutorialMarkDown.md'

import { selectConfigProject } from 'modules/configurationProject/selectors/configurationProjectSelectors'

import { updateConfigProject } from 'modules/configurationProject/actions/configurationProjectActions'

import { GUIDELINES } from 'shared/enums/configurationsProjectPropertiesTypes'

import * as Styled from 'modules/projects/components/__styles__/GuideStepPage.styles'

const GuideStepPage = () => {
	const [tutorialValue, setTutorialValue] = useState(null)
	const [isModeTutorial, setIsModeTutorial] = useState(false)

	const projectConfig = useSelector(selectConfigProject)

	const [guidelinesValue, setGuidelinesValue] = useState(projectConfig?.guidelines || '')

	const dispatch = useDispatch()

	const { t } = useTranslation('configurationProject')

	useEffect(() => {
		fetch(TutorialMarkDown)
			.then((response) => response.text())
			.then((text) => {
				setTutorialValue(text)
			})
	}, [])

	const _onModeChange = (checked) => {
		setIsModeTutorial(checked)
	}

	const _onTextAreaChange = useCallback(
		(e) => {
			setGuidelinesValue(e.target.value)
			!isModeTutorial && dispatch(updateConfigProject(GUIDELINES, { [GUIDELINES]: e.target.value }))
		},
		[isModeTutorial, dispatch]
	)

	return (
		<Styled.Root>
			<Styled.SwitchContainer>
				{t('configurationProject:guide.edit')}
				<Switch checked={isModeTutorial} onChange={_onModeChange} />
				{t('configurationProject:guide.tutorial')}
			</Styled.SwitchContainer>
			<Row gutter={['8', '8']}>
				<Col sm={24} md={12}>
					<Input.TextArea
						value={isModeTutorial ? tutorialValue : guidelinesValue}
						autoSize={{ minRows: 12 }}
						onChange={_onTextAreaChange}
						readOnly={isModeTutorial}
					/>
				</Col>
				<Col sm={24} md={12}>
					<Styled.MarkdownContainer className="markdown-body">
						<ReactMarkdown source={isModeTutorial ? tutorialValue : projectConfig?.guidelines} />
					</Styled.MarkdownContainer>
				</Col>
			</Row>
		</Styled.Root>
	)
}

export default GuideStepPage
