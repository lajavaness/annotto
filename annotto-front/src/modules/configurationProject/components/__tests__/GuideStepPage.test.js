import { Provider } from 'react-redux'
import { ThemeProvider } from 'styled-components'
import { createStore } from 'redux-dynamic-modules'
import { render } from '@testing-library/react'
import React from 'react'

import theme from '__theme__'

import GuideStepPage from 'modules/configurationProject/components/GuideStepPage'

const initialState = {}
const getStore = (state) => createStore({ initialState: state })
const getInstance = (state = initialState) => (
	<Provider store={getStore(state)}>
		<ThemeProvider theme={theme}>
			<GuideStepPage />
		</ThemeProvider>
	</Provider>
)

describe('GuideStepPage', () => {
	it('matches snapshot', () => {
		const { asFragment } = render(getInstance())
		expect(asFragment()).toMatchSnapshot()
	})
})
