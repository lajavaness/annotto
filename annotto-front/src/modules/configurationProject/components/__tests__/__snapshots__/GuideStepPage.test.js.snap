// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GuideStepPage matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-gsBqHK bKJurg"
  >
    <div
      class="ant-space ant-space-horizontal ant-space-align-center sc-aKXVD UYnSr"
    >
      <div
        class="ant-space-item"
        style="margin-right: 8px;"
      >
        configurationProject:guide.edit
      </div>
      <div
        class="ant-space-item"
        style="margin-right: 8px;"
      >
        <button
          aria-checked="false"
          class="ant-switch"
          role="switch"
          type="button"
        >
          <div
            class="ant-switch-handle"
          />
          <span
            class="ant-switch-inner"
          />
        </button>
      </div>
      <div
        class="ant-space-item"
      >
        configurationProject:guide.tutorial
      </div>
    </div>
    <div
      class="ant-row"
      style="margin: -4px -4px -4px -4px;"
    >
      <div
        class="ant-col ant-col-sm-24 ant-col-md-12"
        style="padding: 4px 4px 4px 4px;"
      >
        <textarea
          class="ant-input"
          style="height: 0px; min-height: -48px; max-height: 9007199254740991px; overflow-y: hidden; resize: none; overflow-x: hidden;"
        />
      </div>
      <div
        class="ant-col ant-col-sm-24 ant-col-md-12"
        style="padding: 4px 4px 4px 4px;"
      >
        <div
          class="sc-kmATbt hCQifD markdown-body"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
